File: UD03-Java/Ud02Ex03Radar.java
import java.util.Scanner;

public class Ud02Ex03Radar {
    public static void main(String[] args) {
        final int DISTANCIA = 10;
        final int LIMITE_VELOCIDADE = 120;

        Scanner sc = new Scanner(System.in);

        System.out.println("Introduce el tiempo en segundos:");
        int segundos = sc.nextInt();
        double exceso = excesoVelocidade(DISTANCIA, segundos, LIMITE_VELOCIDADE);
        System.out.println(exceso);
        if (exceso > 0) {
            System.out.println("Infracción por exceso de velocidade!! Multa de "+
            multaInfraccion(LIMITE_VELOCIDADE, exceso) + " €!!");
        } else {
            System.out.println("Velocidade dentro do límite");
        }



/*
        System.out.println(excesoVelocidade(10, 600, 60));
        System.out.println(excesoVelocidade(10, 600, 50));
        System.out.println(excesoVelocidade(10, 600, 30));
        System.out.println(excesoVelocidade(15, 300, 120));
        System.out.println(excesoVelocidade(15, 300, 90));
        System.out.println(excesoVelocidade(10, 300, 120));
        System.out.println(excesoVelocidade(10, 300, 90));

        System.out.println("PRUEBAS MULTAS");
        System.out.println(multaInfraccion(20, 0));
        System.out.println(multaInfraccion(20, 15));
        System.out.println(multaInfraccion(30, 21));
        System.out.println(multaInfraccion(40, 31));
        System.out.println(multaInfraccion(50, 50));
        System.out.println(multaInfraccion(50, 51));

        System.out.println(multaInfraccion(60, 0));
        System.out.println(multaInfraccion(70, 30));
        System.out.println(multaInfraccion(80, 31));
        System.out.println(multaInfraccion(90, 51));
        System.out.println(multaInfraccion(100, 61));
        System.out.println(multaInfraccion(110, 71));
        System.out.println(multaInfraccion(120, 1));*/

    }

    public static double excesoVelocidade(double distancia, int tempo, int limite){
        double exceso = distancia / (tempo / 3600.0) - limite;

        return exceso > 0? exceso:0;
    }


    public static int multaInfraccion(int limite, double excesoVelocidade) {
        int multa = 0;
        if (excesoVelocidade <= 0)
            multa = 0;
        else 
            if (limite <= 50) {
                if (excesoVelocidade <= 20)
                    multa = 100;
                else if (excesoVelocidade <= 30)
                    multa = 300;
                else if (excesoVelocidade <= 40)
                    multa = 400;
                else if (excesoVelocidade <= 50)
                    multa = 500;
                else
                    multa = 600;
            }
            else {
                if (excesoVelocidade <= 30)
                    multa = 100;
                else if (excesoVelocidade <= 50)
                    multa = 300;
                else if (excesoVelocidade <= 60)
                    multa = 400;
                else if (excesoVelocidade <= 70)
                    multa = 500;
                else
                    multa = 600;
            }
        return multa;
    }



}

File: UD02/exame/ud02_ex03_radar.py


def excesoVelocidade(distancia, tempo, limite):
    velocidadeMedia = distancia / (tempo / 3600)
    return 0 if velocidadeMedia <= limite else velocidadeMedia - limite

def puntosInfraccion(limite, excesoVelocidade):
    if limite <= 50:
        if excesoVelocidade <= 20:
            return 0
        elif excesoVelocidade <= 30:
            return 2
        elif excesoVelocidade <= 40:
            return 4
        else:
            return 6
    else:
        if excesoVelocidade <= 30:
            return 0
        elif excesoVelocidade <= 50:
            return 2
        elif excesoVelocidade <= 60:
            return 4
        else:
            return 6

def multaInfraccion(limite, excesoVelocidade):
    if excesoVelocidade <= 0:
        return 0
    elif limite <= 50:
        if excesoVelocidade <= 20:
            return 100
        elif excesoVelocidade <= 30:
            return 300
        elif excesoVelocidade <= 40:
            return 400
        elif excesoVelocidade <= 50:
            return 500
        else:
            return 600
    else:
        if excesoVelocidade <= 30:
            return 100
        elif excesoVelocidade <= 50:
            return 300
        elif excesoVelocidade <= 60:
            return 400
        elif excesoVelocidade <= 70:
            return 500
        else:
            return 600

# PROBAS 
assert excesoVelocidade(10, 600, 60) == 0
assert excesoVelocidade(10, 600, 50) == 10
assert excesoVelocidade(10, 600, 30) == 30
assert excesoVelocidade(15, 300, 120) == 60
assert excesoVelocidade(15, 300, 90) == 90
assert excesoVelocidade(10, 300, 120) == 0
assert excesoVelocidade(10, 300, 90) == 30

assert multaInfraccion(20, 0) == 0
assert multaInfraccion(20, 15) == 100
assert multaInfraccion(30, 21) == 300
assert multaInfraccion(40, 31) == 400
assert multaInfraccion(50, 50) == 500
assert multaInfraccion(50, 51) == 600

assert multaInfraccion(60, 0) == 0
assert multaInfraccion(70, 30) == 100
assert multaInfraccion(80, 31) == 300
assert multaInfraccion(90, 51) == 400
assert multaInfraccion(100, 61) == 500
assert multaInfraccion(110, 71) == 600
assert multaInfraccion(120, 1) == 100

